bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/ZkLedgerUnderreplicationManager.java

```diff
@@ -393,11 +390,17 @@ public boolean hasNext() {
                                 String child = parent + "/" + c;
                                 if (c.startsWith("urL")) {
                                     long ledgerId = getLedgerId(child);
-                                    List<String> replicaList = getLedgerUnreplicationInfo(ledgerId).getReplicaList();
-                                    if ((predicate == null)
-                                            || predicate.test(replicaList)) {
-                                        curBatch.add(new AbstractMap.SimpleImmutableEntry<Long, List<String>>(ledgerId,
-                                                ((includeReplicaList) ? replicaList : null)));
+                                    UnderreplicatedLedgerFormat underreplicatedLedgerFormat =
+                                            getLedgerUnreplicationInfo(ledgerId);
+                                    List<String> replicaList = underreplicatedLedgerFormat.getReplicaList();
+                                    long ctime = (underreplicatedLedgerFormat.hasCtime()
+                                            ? underreplicatedLedgerFormat.getCtime() : -1L);
```

Comment:
What about having a constant for -1 (unknown) ?

Reply:
done